bootloader/src/pyi_utils.c

```diff
@@ -632,17 +666,34 @@ static int
 set_dynamic_library_path(const char* path)
 {
     int rc = 0;
+    char *env_var, *env_var_orig;
+    char *new_path, *orig_path;
 
     #ifdef AIX
     /* LIBPATH is used to look up dynamic libraries on AIX. */
-    pyi_setenv("LIBPATH", path);
-    VS("LOADER: LIBPATH=%s\n", path);
+    env_var = "LIBPATH";
+    env_var_orig = "LIBPATH_ORIG";
     #else
     /* LD_LIBRARY_PATH is used on other *nix platforms (except Darwin). */
-    rc = pyi_setenv("LD_LIBRARY_PATH", path);
-    VS("LOADER: LD_LIBRARY_PATH=%s\n", path);
+    env_var = "LD_LIBRARY_PATH";
+    env_var_orig = "LD_LIBRARY_PATH_ORIG";
     #endif /* AIX */
 
+    /* keep original value in a new env var so the application can restore it
+     * before forking subprocesses. This is important so that e.g. a forked
+     * (system installed) ssh can find the matching (system installed) ssh
+     * related libraries - not the potentially different versions of same libs
+     * that we have bundled.
+     */
+    orig_path = pyi_getenv(env_var);
+    pyi_setenv(env_var_orig, orig_path);
+    VS("LOADER: %s=%s\n", env_var_orig, orig_path);
+
+    /* prepend our path to the original path */
+    new_path = pyi_strjoin(path, ":", orig_path);
```

Comment:
How about this?

```c
char* new_path = alloc(path + 1 + orig_path + 1);
sprintf(new_path, "%s:%s", path, orig_path);
```

I think this is the much simpler solution. (the double +1 could just be a +2, but I put it this way to signify why it's there)